File: app/controllers/dto/reporting/BudgetReportEntry.scala
package controllers.dto.reporting

import java.time.LocalDate
import play.api.libs.json._

import controllers.dto.BudgetPair
import controllers.dto.BudgetDTO.budgetPairWrites //Used implicitly by Json.writes below

case class BudgetReportEntry(date: LocalDate, income: BudgetPair, expense: BudgetPair, profit: BigDecimal) extends ReportValue;

object BudgetReportEntry {
  implicit val budgetReportEntryWrite = Json.writes[BudgetReportEntry]
}

File: backend/src/main/java/org/akashihi/mdg/entity/report/BudgetReportEntry.java
package org.akashihi.mdg.entity.report;

import org.akashihi.mdg.entity.Budget;

import java.math.BigDecimal;
import java.time.LocalDate;

public record BudgetReportEntry(LocalDate date, Budget.BudgetPair income, Budget.BudgetPair expense, BigDecimal profit) { }
